presto-main/src/main/java/com/facebook/presto/sql/planner/GroupingOperationRewriter.java

```diff
@@ -50,14 +47,14 @@
     private final QuerySpecification queryNode;
     private final Analysis analysis;
     private final Metadata metadata;
-    private final Map<QuerySpecification, Symbol> groupIdSymbols;
+    private final Optional<Symbol> groupIdSymbol;
 
-    public GroupingOperationRewriter(QuerySpecification queryNode, Analysis analysis, Metadata metadata, Map<QuerySpecification, Symbol> groupIdSymbols)
+    public GroupingOperationRewriter(QuerySpecification queryNode, Analysis analysis, Metadata metadata, Optional<Symbol> groupIdSymbol)
     {
         this.queryNode = requireNonNull(queryNode, "node is null");
         this.analysis = requireNonNull(analysis, "analysis is null");
         this.metadata = requireNonNull(metadata, "metadata is null");
-        this.groupIdSymbols = requireNonNull(groupIdSymbols, "groupIdSymbols is null");
+        this.groupIdSymbol = requireNonNull(groupIdSymbol, "groupIdSymbol is null");
```

Comment:
Nope there's no need for it to be an `ExpressionVisitor`. I was stuck in my old solution. Changed.